File: fe/src/api/feedback.ts
import { request } from './client'

// A "this result is right" mark from the smart-search proof card. Stored in this
// project's DB and used only as few-shot to steer the LLM reranker (project-local
// learning — the shared model is never trained). `resource` scopes it to a screen.
export type FeedbackCreate = {
  resource: 'devices' | 'maintenance' | 'handovers'
  item_id: string
  query: string
  document?: string
  score?: number
  label?: number // 1 = right / relevant, 0 = not relevant
}

export function markFeedback(fb: FeedbackCreate) {
  return request('/feedback', {
    method: 'POST',
    body: JSON.stringify({ label: 1, ...fb }),
  })
}

export function feedbackStats() {
  return request<{ total: number }>('/feedback/stats')
}

File: be/models/feedback.py
"""Search-feedback schemas — the per-project signal that steers the LLM reranker.

These rows are stored in THIS project's DB and used only as few-shot anchors in
this project's prompts; the shared LLM is never trained. `resource` scopes a mark
to one screen (devices / maintenance / handovers).
"""
from datetime import datetime

from pydantic import BaseModel


class FeedbackCreate(BaseModel):
    resource: str  # "devices" | "maintenance" | "handovers"
    item_id: str
    query: str
    document: str | None = None
    score: float | None = None
    label: int = 1  # 1 = correct/relevant, 0 = not relevant


class FeedbackOut(FeedbackCreate):
    id: int
    created_at: datetime
